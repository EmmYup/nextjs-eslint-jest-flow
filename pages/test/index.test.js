import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../index';

describe('With Enzyme', () => {
  it('App shows "Welcome to next.js with Eslint, Jest and Flow!"', () => {
    const app = shallow(<App />);

    expect(app.find('p').text()).toEqual(
      'Welcome to next.js with Eslint, Jest and Flow!'
    );
  });
});

describe('With Snapshot Testing', () => {
  it('App shows "Welcome to next.js with Eslint, Jest and Flow!"', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
