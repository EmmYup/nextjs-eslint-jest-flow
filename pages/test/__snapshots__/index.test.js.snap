// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Welcome to next.js with Eslint, Jest and Flow!" 1`] = `
<div>
   
  <p>
    Welcome to next.js with Eslint, Jest and Flow!
  </p>
   
</div>
`;
